language_service/src/completion.rs

```diff
@@ -19,11 +19,49 @@ const PRELUDE: [&str; 3] = [
     "Microsoft.Quantum.Intrinsic",
 ];
 
+struct IndentMaker<'a> {
+    contents: &'a str,
+    offset: u32,
+}
+
+impl<'a> IndentMaker<'a> {
+    fn new(compilation: &'a Compilation, source_name: &'_ str) -> Self {
+        let source = compilation
+            .user_unit()
+            .sources
+            .find_by_name(source_name)
```

Comment:
Since you already have the package offset for the cursor, it's going to be more efficient if you look up the source contents with `find_by_offset`, instead of looking up using the name.

Or maybe you can have `source_offset_to_package_offset` return the `SourceContents` as well? Since it's getting called in the next line anyway.

Reply:
done